test/dotnet-watch.FunctionalTests/DotNetWatchTests.cs

```diff
@@ -0,0 +1,266 @@
+            }
+        }
+
+        [Fact(Skip = "Blocked on https://github.com/aspnet/dnx-watch/issues/4")]
+        public void ExitOnChange()
+        {
+            var scenario = new ProjectToolScenario();
+            scenario.AddNugetFeed(DotnetWatch, _artifactsFolder);
+            scenario.AddProject(_testAppFolder);
+            scenario.AddToolToProject(TestAppName, DotnetWatch);
+            scenario.Restore();
+
+            var statusFilePath = Path.Combine(scenario.TempFolder, "status");
+            var startedFile = statusFilePath + ".started";
+
+            Process watcherProcess = null;
+
+            try
+            {
+                // Wait for the process to start
+                using (var wait = new WaitForFileToChange(startedFile))
+                {
+                    watcherProcess = scenario.ExecuteDotnet($"watch --exit-on-change -- {statusFilePath} --no-exit", Path.Combine(scenario.WorkFolder, TestAppName));
+
+                    if (!wait.Wait(_defaultTimeout))
+                    {
+                        throw new Exception($"File not created: {startedFile}");
+                    }
+                }
+
+                // Change a file
+                var fileToChange = Path.Combine(scenario.WorkFolder, TestAppName, "Program.cs");
+                var programCs = File.ReadAllText(fileToChange);
+                File.WriteAllText(fileToChange, programCs);
+
+                Console.WriteLine($"Waiting for the watcher ({watcherProcess.Id}) to exit...");
+                // Now the watcher should exit
+                Assert.True(!watcherProcess.WaitForExit((int)_defaultTimeout.TotalMilliseconds));
+
+                // Make sure the child process if dead
+                using (var reader = File.OpenText(startedFile))
+                {
+                    var firstProcessId = int.Parse(reader.ReadLine());
+                    Console.WriteLine($"Checking that PID {firstProcessId} doesn't exist anymore.");
+
+                    try
+                    {
+                        var process = Process.GetProcessById(firstProcessId);
+                        Assert.True(process.WaitForExit((int)_defaultTimeout.TotalMilliseconds));
+                    }
+                    catch (ArgumentException)
+                    {
+                        // The process is gone so we're good
+                    }
+                }
+            }
+            finally
+            {
+                if (watcherProcess != null)
+                {
+                    if (!watcherProcess.HasExited)
+                    {
+                        KillProcess(watcherProcess?.Id);
+                    }
+                    watcherProcess.Dispose();
+                }
+                scenario.Dispose();
+            }
+        }
+
+        private static string FindRepoRoot()
+        {
+            var currentDirPath = Directory.GetCurrentDirectory();
+
+            string globalJsonFile;
+            while (true)
```

Comment:
shuldn't you break out of this loop (possibly with an exception) if you reach the folder root?